log/log.go

```diff
@@ -1,72 +1,129 @@
 package log
 
 import (
-	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
 
-	"github.com/golang/glog"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
 )
 
-const (
-	debug glog.Level = glog.Level(4)
-	trace glog.Level = glog.Level(5)
-)
+// Configures the global log level and log format.
+func InitializeLogger() zerolog.Logger {
+	logTimeFieldFormat, isTimeFieldFormatDefined := os.LookupEnv("LOG_TIME_FIELD_FORMAT")
+
+	if !isTimeFieldFormatDefined {
+		logTimeFieldFormat = time.RFC3339
+	}
```

Comment:
Is there some way to parse the time field format for correctness? Similar to what you do for LOG_SAMPLER_RATE down below when you confirm it can be parsed as a number, and what you do for LOG_LEVEL in resolveLogLevelFromEnv().

Perhaps zerolog has a way to check it for validity. I'm thinking what happens if a person enters some invalid time field - we would want this to either fail-fast or default to some valid format. Something like:
```suggestion
	} else if !zerolog.IsTimeFieldFormatValid(logTimeFieldFormat) { // I have no idea if that API exists - but you get the idea
		log.Warn().Msgf("Provided time field format [%s] is not valid. Defaulting to RFC3339.", logTimeFieldFormat)
		logTimeFieldFormat = time.RFC3339
	}
```

Reply:
Hm it does rely on the standard time package (https://golang.org/pkg/time/) which to the best of my knowledge could take any malformed time format and parse it to a non-sense time object. 

I tried to do a couple of format checks on the current timestamp ála:
```
timestamp := time.Now()
parsedAsString := timestamp.Format(format)
parsedAsTimestamp, err := time.Parse(format, parsedAsString)
```
but this seems to be unlikely successful as the format can range from `stupidInput`, `2006` to a standard format like `2006-01-02T15:04:05Z07:00`. Especially the fact that you can provide wrong formats like `2020` which will completely mess up the logic (parses the year to 9090) will probably brake the check routine. 

Neither the time module itself nor zerolog has validators for the format. What we might want to consider is that we only support a subset of time formats like the ones shipped with the time module:
```
	ANSIC       = "Mon Jan _2 15:04:05 2006"
	UnixDate    = "Mon Jan _2 15:04:05 MST 2006"
	RubyDate    = "Mon Jan 02 15:04:05 -0700 2006"
	RFC822      = "02 Jan 06 15:04 MST"
	RFC822Z     = "02 Jan 06 15:04 -0700" // RFC822 with numeric zone
	RFC850      = "Monday, 02-Jan-06 15:04:05 MST"
	RFC1123     = "Mon, 02 Jan 2006 15:04:05 MST"
	RFC1123Z    = "Mon, 02 Jan 2006 15:04:05 -0700" // RFC1123 with numeric zone
	RFC3339     = "2006-01-02T15:04:05Z07:00"
	RFC3339Nano = "2006-01-02T15:04:05.999999999Z07:00"
	Kitchen     = "3:04PM"
	// Handy time stamps.
	Stamp      = "Jan _2 15:04:05"
	StampMilli = "Jan _2 15:04:05.000"
	StampMicro = "Jan _2 15:04:05.000000"
	StampNano  = "Jan _2 15:04:05.000000000"
```

We could check against those and fallback to `RFC3339` in case the validations fails. What do you think?

Reply:
Proposal see below.